seatunnel-e2e/seatunnel-connector-v2-e2e/connector-file-sftp-e2e/src/test/java/org/apache/seatunnel/e2e/connector/file/fstp/SftpFileIT.java

```diff
@@ -0,0 +1,210 @@
+                container.executeJob("/text/sftp_file_text_projection_to_assert.conf");
+        Assertions.assertEquals(0, textProjectionResult.getExitCode());
+        // test write sftp json file
+        Container.ExecResult jsonWriteResult =
+                container.executeJob("/json/fake_to_sftp_file_json.conf");
+        Assertions.assertEquals(0, jsonWriteResult.getExitCode());
+        // test read sftp json file
+        Container.ExecResult jsonReadResult =
+                container.executeJob("/json/sftp_file_json_to_assert.conf");
+        Assertions.assertEquals(0, jsonReadResult.getExitCode());
+    }
+
+    public static void initDate(String host, int mappedPort) {
+        Map<String, String> filesMap = new HashMap<>();
+        filesMap.put(
+                "/excel/e2e.xlsx", "/tmp/seatunnel/read/excel/name=tyrantlucifer/hobby=coding/");
+        filesMap.put("/json/e2e.json", "/tmp/seatunnel/read/json/name=tyrantlucifer/hobby=coding/");
+        filesMap.put("/text/e2e.txt", "/tmp/seatunnel/read/text/name=tyrantlucifer/hobby=coding/");
+
+        for (Map.Entry<String, String> entry : filesMap.entrySet()) {
+            String localPath = entry.getKey();
+            String remotePath = entry.getValue();
+            Path path = ContainerUtil.getResourcesFile(localPath).toPath();
+
+            tmpFileToSftpServerWithCustomPath(
+                    host, mappedPort, USERNAME, PASSWORD, path.toFile(), remotePath);
+        }
+    }
+
+    public static void tmpFileToSftpServerWithCustomPath(
+            String host,
+            int port,
+            String username,
+            String password,
+            File localFilePath,
+            String remotePath) {
+        JSch jsch = new JSch();
+        Session session = null;
+        ChannelSftp channelSftp = null;
+
+        try {
+            session = jsch.getSession(username, host, port);
+            session.setPassword(password);
+            java.util.Properties config = new java.util.Properties();
+            config.put("StrictHostKeyChecking", "no");
+            config.put("PreferredAuthentications", "password");
+            session.setConfig(config);
+            session.connect();
+
+            channelSftp = (ChannelSftp) session.openChannel("sftp");
+            channelSftp.connect();
+
+            StringTokenizer token = new StringTokenizer(remotePath, "/");
+            String path = "";
+
+            while (token.hasMoreTokens()) {
+                path = path + "/" + token.nextToken();
+                try {
+                    channelSftp.cd(path);
+                } catch (SftpException e) {
+                    channelSftp.mkdir(path);
+                    channelSftp.cd(path);
+                }
+            }
+
+            FileInputStream fis = new FileInputStream(localFilePath);
+
+            channelSftp.put(fis, localFilePath.getName());
+
+            fis.close();
+        } catch (Exception e) {
+            e.printStackTrace();
```

Comment:
Assertions.assertThrows is used to test and ensure that a method throws the expected exception under specific conditions, which might not be quite suitable for this logic. I think it would be better to modify it like this:
log.error("File to sftpServer with custom path operation exception", e);